recipes/dxpr_cms_image_media_type/recipe.yml

```diff
@@ -6,9 +6,15 @@ recipes:
 install:
   - focal_point
   - media_file_delete
+  - key
+  - ai
+  - provider_openai
+  - provider_anthropic
```

Comment:
I can see how we need to enable this for the key check during installation, but I'm inclined to think would be better if we:
1. Enable them programmatically before the Keys profile page loads
2. Disable whichever is/are not used in the cleanup step:
 https://github.com/dxpr/dxpr_cms/blob/e9b9e5f6146cb568c7f9d13a28a90922a9c7ba6c/installer/dxpr_cms_installer.profile#L159-L166

I might be wrong, let me know your thoughts

Reply:
It would make sense to uninstall the not used one as a last step, because then I could use the actual installed provider for doing the authentication check and get a list of available models in the provider. The AI module has the same problem, that we need to hardcode this date as a config, because it's not provided via the API from Anthropic, but at least we will keep an eye out for if it changes (and I'll add a [test](https://www.drupal.org/project/ai/issues/3469650) for it)

Reply:
@ivanboring  This is updated now, with the above mentioned updates. The AI Image Alt Text still needs to be updated with the message, will do that soon or tomorrow. There are two notes:

1. If No AI is chosen, it still keeps both providers installed, it only uninstalls the ones who are not being used, if one is installed. Correct?
2. The recipe that creates the roles has the media recipe as dependency, which means that the permission can't be set without causing circular dependencies. I've added so it gets created if its missing, which solves it for DXPR, but it also means that anyone that installs the recipe outside of DXPR, gets these roles. Correct?

Reply:
yes